tests/signing/in_toto_signature_test.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+from model_signing.hashing import file
+from model_signing.hashing import memory
+from model_signing.serialization import serialize_by_file
+from model_signing.serialization import serialize_by_file_shard
+from model_signing.signature import fake
+from model_signing.signing import in_toto
+from model_signing.signing import in_toto_signature
+
+
+class TestIntotoSignature:
+    def _shard_hasher_factory(
+        self, path: pathlib.Path, start: int, end: int
+    ) -> file.ShardedFileHasher:
+        return file.ShardedFileHasher(
+            path, memory.SHA256(), start=start, end=end
+        )
+
+    def _hasher_factory(self, path: pathlib.Path) -> file.FileHasher:
+        return file.SimpleFileHasher(path, memory.SHA256())
+
+    def test_to_manifest(self, sample_model_folder):
+        signer = in_toto_signature.IntotoSigner(fake.FakeSigner())
+        shard_serializer = serialize_by_file_shard.ManifestSerializer(
+            self._shard_hasher_factory, allow_symlinks=True
+        )
+        shard_manifest = shard_serializer.serialize(sample_model_folder)
+
+        for payload_type in [
+            in_toto.ShardDigestsIntotoPayload,
+            in_toto.DigestOfShardDigestsIntotoPayload,
+        ]:
+            payload = payload_type.from_manifest(shard_manifest)
+            print(payload.statement.pb.predicate_type)
+            sig = signer.sign(payload)
+            manifest = sig.to_manifest()
+            assert shard_manifest == manifest
+
+        file_serializer = serialize_by_file.ManifestSerializer(
+            self._hasher_factory, allow_symlinks=True
+        )
+        file_manifest = file_serializer.serialize(sample_model_folder)
+        for payload_type in [
+            in_toto.DigestOfDigestsIntotoPayload,
+            in_toto.DigestsIntotoPayload,
+        ]:
+            payload = payload_type.from_manifest(file_manifest)
+            sig = signer.sign(payload)
+            manifest = sig.to_manifest()
+            assert file_manifest == manifest
```

Comment:
Please separate each test case. Right now if one fails the others are not tested.